subprojects/core/src/main/java/org/gradle/api/internal/file/copy/DefaultCopySpec.java

```diff
@@ -71,7 +71,10 @@
     private Integer fileMode;
     private Boolean caseSensitive;
     private Boolean includeEmptyDirs;
-    private DuplicatesStrategy duplicatesStrategy = DuplicatesStrategy.INHERIT;
+
+    private static final DuplicatesStrategy DEFAULT_DUPLICATION_STRATEGY = DuplicatesStrategy.INHERIT;
```

Comment:
`INHERIT` will delegate to the parent spec. If there is no parent spec, or the parent spec did not set something explicitly, then the default is returned - `INCUDE`. So the variable name is actually misleading: The default duplicate strategy is `INCLUDE`, not `INHERIT`. `INHERIT` serves as a null-object to make it easier to use the DSL from Kotlin.

So I would suggest you inline the constant or use something like this:
```
private static final DuplicatesStrategy DEFAULT_DUPLICATES_STRATEGY = DuplicatesStrategy.INCLUDE;

...

        public DuplicatesStrategy getDuplicatesStrategy() {
            if (duplicatesStrategy != DuplicatesStrategy.INHERIT) {
                return duplicatesStrategy;
            }
            if (parentResolver != null) {
                return parentResolver.getDuplicatesStrategy();
            }
            return DEFAULT_DUPLICATES_STRATEGY;
        }
```

Reply:
Inlined, thanks for explaining (it's why I wanted to use `null` previously as a default, to signal that it wasn't set yet)